TodoApi/Extensions/SerilogExtensions.cs

```diff
@@ -0,0 +1,132 @@
+using System.Globalization;
+using System.Reflection;
+using Microsoft.Extensions.Options;
+using Serilog;
+using Serilog.Events;
+using Serilog.Exceptions;
+using Serilog.Sinks.Elasticsearch;
+
+namespace TodoApi;
+
+public static class SerilogExtensions
+{
+    public static WebApplicationBuilder AddSerilog(
+        this WebApplicationBuilder builder,
+        string sectionName = "Serilog",
+        Action<LoggerConfiguration>? extraConfigure = null)
+    {
+        builder.Services.AddOptions<SerilogOptions>()
+            .BindConfiguration(sectionName)
+            .ValidateDataAnnotations();
+
+
+        builder.Host.UseSerilog((context, serviceProvider, loggerConfiguration) =>
+        {
+            var loggerOptions = serviceProvider.GetRequiredService<IOptions<SerilogOptions>>().Value;
+
+            // https://github.com/serilog/serilog-settings-configuration
+            loggerConfiguration.ReadFrom.Configuration(context.Configuration, sectionName: sectionName);
+
+            extraConfigure?.Invoke(loggerConfiguration);
+
+            var levelString = context.Configuration.GetValue<string>($"{sectionName}:MinimumLevel:Default") ??
+                        nameof(LogEventLevel.Information);
+
+            Enum.TryParse<LogEventLevel>(levelString, true, out var logLevel);
```

Comment:
This feels a bit too manual and custom. Serilog doesn't have a "default from configuration experience"?

Reply:
This `loggerConfiguration.ReadFrom.Configuration` is for reading serilog configs from appsettings, but moving all settings to appsettings makes it bigger

Reply:
Can you explain?